lib/tesla/adapter/gun.ex

```diff
@@ -356,13 +397,16 @@ if Code.ensure_loaded?(:gun) do
           check_body_size(acc, body, limit)
 
         {:gun_data, ^pid, ^stream, :nofin, part} ->
-          case check_body_size(acc, part, limit) do
-            {:ok, acc} -> read_body(pid, stream, opts, acc)
-            {:error, error} -> {:error, error}
+          with {:ok, acc} <- check_body_size(acc, part, limit) do
+            read_body(pid, stream, opts, acc)
           end
+
+        {:DOWN, _, _, _, reason} ->
+          if opts[:close_conn], do: close(pid)
+          {:error, reason}
       after
         opts[:timeout] || @adapter_default_timeout ->
-          {:error, :timeout}
+          {:error, :recv_body_timeout}
```

Comment:
No close here?

Reply:
yes, this `:error` case will be closed in `format_response/6` with `:plain` clause. Also i removed unnecessary close check in `{:DOWN, _, _, _, reason}` clause.